tests/post/steps/test_service_configuration.py

```diff
@@ -0,0 +1,261 @@
+import yaml
+
+import pytest
+from pytest_bdd import scenario, given, then, when, parsers
+
+import kubernetes.client
+from kubernetes.client import AppsV1Api
+from kubernetes.client.rest import ApiException
+
+from tests import utils
+
+
+# Constants {{{
+
+
+BODY = """
+data:
+  config.yaml: |-
+    apiVersion: addons.metalk8s.scality.com
+    kind: DexConfig
+    spec:
+      deployment:
+        replicas: {replicas_count}
+      localuserstore:
+        enabled: true
+        userlist:
+          - email: "admin@metalk8s.invalid"
+            hash: "$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W"
+            username: "admin"
+            userID: "08a8684b-db88-4b73-90a9-3cd1661f5466"
+      connectors: []
+"""
+
+
+# }}}
+
+
+# Fixtures {{{
+
+
+@pytest.fixture
+def k8s_appsv1_client(k8s_apiclient):
+    return AppsV1Api(api_client=k8s_apiclient)
+
+@pytest.fixture
+def teardown(host, version, request, k8s_client):
+    yield
+    teardown_configmap(k8s_client)
+
+
+# }}}
+
+
+# Scenarios {{{
+
+
+@scenario('../features/service_configuration.feature',
+          'Propagation of Service Configurations to underlying Services')
+def test_service_config_propagation(host, teardown):
+    pass
+
+
+# }}}
+
+
+# Given {{{
+
+
+@given(parsers.parse(
+    "we have a '{name}' CSC in namespace '{namespace}' with "
+    "'{path}' equal to '{value}'"))
+def check_configmap_configuration(
+    host,
+    k8s_client,
+    name,
+    namespace,
+    path,
+    value
+):
+    response = read_namespace_config_map(k8s_client, name, namespace)
+
+    assert response, ("No ConfigMap with name {} found".format(name))
+
+    try:
+        response_yaml = yaml.safe_load(response.data['config.yaml'])
+    except yaml.YAMLError as exc:
+        raise Exception(
+            'Invalid YAML format in ConfigMap {}: {!s}'.format(name, exc)
+        )
+    except Exception as exc:
+        raise Exception(
+            'Failed loading `config.yaml` from ConfigMap {}: {!s}'.format(
+                name, exc
+            )
+        )
+
+    response_value = \
+        eval("response_yaml"+"{}".format(path))
+
+    assert int(value) == response_value, (
+        "Expected value {} for key {}, got {}".format(
+            value, path, response_value
+        )
+    )
+
+
+# }}}
+
+
+# When {{{
+
+
+@when(parsers.parse(
+    "we update '{name}' CSC in namespace '{namespace}' "
+    "'{path}' to '{value}'"))
+def update_service_configuration(
+    host,
+    k8s_client,
+    name,
+    namespace,
+    path,
+    value
+):
+    # This patch here overrides the entire config.yaml
+    try:
+        response = k8s_client.patch_namespaced_config_map(
+            name, namespace, yaml.safe_load(BODY.format(
+                replicas_count=value)
+            )
+        )
```

Comment:
No please do not override the full config,
Retrieve it load the yaml and edit the field you want and then push it back